### DISCUSSION TITLE
How will lockup contract be deployed?

### DISCUSSION THREAD

**Discussion by alexaunter** [galactic3/ft-lockup-contract#2] [Q&A]
How will this new ft lockup contract be deployed?
(in current implementation there is a factory generating lockup contracts as sub-accounts for lockup.near through factory)
It seems like nothing is blocking us from adopting the same approach.

Are there any other important details (for example in context of wallet integration) regarding the deployment?

**Comment by AngelBlock** [comment]
The high level question is: is there a plan to have a standardised and supported by wallet/explorer account for a new lockups similar to `lockup.near`?

**Comment by alexaunter** [comment]
Quoting @frol:
We can adopt previous approach: 
- factory that creates sub-accounts for lockups, 
- one lockup sub-account locks-up single token type, possibly for multiple users
- two separate lockups for same token can be created, for different owners
- for wallet integration we can use indexer to look up all lockup contracts for specific user account
